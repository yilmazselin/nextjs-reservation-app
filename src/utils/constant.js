export const localeOptions = [
  {
    id: 0,
    key: "tr"
  },
  {
    id: 1,
    key: "en"
  }
]
