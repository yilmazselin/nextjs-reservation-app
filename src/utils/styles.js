export const padding = {
  xs: 8,
  sm: 16,
  md: 24,
  lg: 32,
  xl: 40,
};

export const fontSize = {
  sm: 12,
  nm: 14,
  md: 16,
  lg: 24,
  xl: 32,
};

export const colors = {
  black: "#E82223",
  dark: "#121212",
  gray: "#64738C",
  light: "#E6E6E6",
  red: "#E82223",
  green: "#0DAFC0",
  yellow: "#ECD03F",
  blue: "#0078E3",
  white: "#fff",
  disabled: "#BBC3CF",
  zero: "#000000",
};

export const fontWeight = {
  bold: 700,
  semiBold: 600,
  regular: 500,
  medium: 400,
  light: 300,
};
