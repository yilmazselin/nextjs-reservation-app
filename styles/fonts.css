
    /*! Generated by Font Squirrel (https://www.fontsquirrel.com) on September 6, 2022 */



@font-face {
    font-family: 'NunitoSans-700';
    src: url('../assets/fonts/NunitoSans/NunitoSans-Bold.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;

}

@font-face {
    font-family: 'NunitoSans-600';
    src: url('../assets/fonts/NunitoSans/NunitoSans-SemiBold.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;
}


@font-face {
    font-family: 'NunitoSans-500';
    src: url('../assets/fonts/NunitoSans/NunitoSans-Regular.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;

}



@font-face {
    font-family: 'NunitoSans-400';
    src: url('../assets/fonts/NunitoSans/NunitoSans-Light.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;

}
@font-face {
    font-family: 'NunitoSans-300';
    src: url('../assets/fonts/NunitoSans/NunitoSans-ExtraLight.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;

}



