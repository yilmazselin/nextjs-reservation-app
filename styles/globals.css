*, *:before, *:after {
    box-sizing: border-box;
  }
  html, body, div, span, applet, object, iframe, h1, h2, h3, h4, h5, h6, p, blockquote, pre, a, abbr, acronym, address, big, cite, code, del, dfn, em, img, ins, kbd, q, s, samp, small, strike, strong, sub, sup, tt, var, b, u, i, center, dl, dt, dd, ol, ul, li, fieldset, form, label, legend, table, caption, tbody, tfoot, thead, tr, th, td, article, aside, canvas, details, embed, figure, figcaption, footer, header, hgroup, menu, nav, output, ruby, section, summary, time, mark, audio, video {
    margin: 0;
    padding: 0;
    border: 0;
    font-size: 100%;
    vertical-align: baseline;
    user-select: none;
  }
  article, aside, details, figcaption, figure, footer, header, hgroup, menu, nav, section {
    display: block;
  }
  body {
    line-height: 1;
  }
  ol, ul {
    list-style: none;
  }
  blockquote, q {
    quotes: none;
  }
  blockquote:before, blockquote:after {
    content: '';
    content: none;
  }
  q:before, q:after {
    content: '';
    content: none;
  }
  table {
    border-collapse: collapse;
    border-spacing: 0;
  }
  button {
    background-color: transparent;
    outline: none;
    border: 0;
    cursor: pointer;
  }
  ol, ul, li {
    list-style: none;
  }
  a {
    text-decoration: none;
  }

  
body {
    background-color: #fff;
    font-family: "NunitoSans-400", sans-serif;
    font-size: 14px;
    -webkit-font-smoothing: antialiased
}
.loader {
  font-size: 10px;
  text-indent: -9999em;
  width: 24px;
  height: 24px;
  border-radius: 50%;
  background: #ffffff;
  background: -moz-linear-gradient(left, #ffffff 10%, rgba(255, 255, 255, 0) 42%);
  background: -webkit-linear-gradient(left, #ffffff 10%, rgba(255, 255, 255, 0) 42%);
  background: -o-linear-gradient(left, #ffffff 10%, rgba(255, 255, 255, 0) 42%);
  background: -ms-linear-gradient(left, #ffffff 10%, rgba(255, 255, 255, 0) 42%);
  background: linear-gradient(to right, #ffffff 10%, rgba(255, 255, 255, 0) 42%);
  position: relative;
  -webkit-animation: load3 1.4s infinite linear;
  animation: load3 1.4s infinite linear;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}
.loader:before {
  width: 50%;
  height: 50%;
  background: #ffffff;
  border-radius: 100% 0 0 0;
  position: absolute;
  top: 0;
  left: 0;
  content: '';
}
.loader:after {
  background: #BBC3CF;
  width: 75%;
  height: 75%;
  border-radius: 50%;
  content: '';
  margin: auto;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}
@-webkit-keyframes load3 {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}
@keyframes load3 {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}
