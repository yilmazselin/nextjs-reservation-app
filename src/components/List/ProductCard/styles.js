import styled from "styled-components";
import { colors, fontSize, fontWeight, padding } from "../../../utils/styles";

export const Container = styled.div`
  border-bottom: 1px solid #f1f3f5;
  padding-top: ${padding.sm}px;
  padding-bottom: ${padding.sm}px;
`;

export const SubContainer = styled.div`
  display: flex;
`;

export const ProductImageWrapper = styled.div`
  height: 75px;
  position: relative;
  display: inline-block;
  border-radius: 12px;
  filter: drop-shadow(0px 10px 15px #c8c9cd);
  img {
    display: block;
    width: auto;
    border-radius: 12px;
  }
  &:after {
    content: "";
    border-radius: 12px;
    position: absolute;
    left: 0%;
    top: 0px;
    width: 100%;
    height: 100%;
    display: inline-block;
    background: linear-gradient(
      180deg,
      rgba(0, 0, 0, 0) 41.03%,
      rgba(0, 0, 0, 0) 41.03%,
      rgba(0, 0, 0, 0.8) 84.62%
    );
  }
`;

export const ProductInfoWrapper = styled.div`
  margin-left: ${padding.sm}px;
`;

export const Row = styled.div`
  display: flex;
  align-items: center;

  &:nth-child(2) {
    margin-left: ${padding.md}px;
  }
`;
export const PriceWrapper = styled.div`
  margin-top: ${padding.xs + 2}px;
`;

export const AddBasket = styled.button`
  margin-top: ${padding.xs + 2}px;
  display: flex;
  align-items: center;
  margin-left: auto;
`;

export const styles = {
  buttonTextStyle: {
    fontSize: fontSize.sm,
    fontWeight: fontWeight.bold,
    color: colors.blue,
    params: {
      letterSpacing: "0.1px",
      marginLeft: padding.xs,
      textTransform: "uppercase",
    },
  },
  smallTextStyle: {
    fontSize: fontSize.sm,
    fontWeight: fontWeight.regular,
    color: colors.dark,
    params: {
      letterSpacing: "0.1px",
    },
  },
  productSubTitleStyle: {
    fontSize: fontSize.sm,
    fontWeight: fontWeight.regular,
    color: colors.gray,
    params: {
      letterSpacing: "0.1px",
      marginTop: padding.xs + 2,
      marginBottom: padding.xs + 2,
    },
  },
  productTitleStyle: {
    fontSize: fontSize.md,
    fontWeight: fontWeight.semiBold,
    color: colors.dark,
    params: {
      letterSpacing: "0.1px",
    },
  },
};
