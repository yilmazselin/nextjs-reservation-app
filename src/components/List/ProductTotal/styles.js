import styled from "styled-components";
import { colors, fontSize, fontWeight, padding } from "../../../utils/styles";

export const Container = styled.div`
  margin-top: 40px;
  margin-bottom: 33px;
`;

export const styles = {
  bigTitleStyle: {
    color: colors.zero,
    fontWeight: fontWeight.bold,
    fontSize: fontSize.lg,
    params: {
      marginBottom: padding.md,
    },
  },
  secondaryTextStyle: {
    color: colors.zero,
    fontWeight: fontWeight.regular,
    fontSize: fontSize.nm,
    params: {
      marginBottom: padding.sm,
    },
  },
  subTitleStyle: {
    color: colors.zero,
    fontSize: fontSize.md,
    fontWeight: fontWeight.bold,
    params: {
      marginTop: padding.md,
    },
  },
};
