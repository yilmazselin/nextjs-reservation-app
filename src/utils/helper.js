import { css } from "styled-components"
import { colors, fontSize, fontWeight } from "./styles"

const breakpoints = {
  sm: 576,
  md: 768,
  lg: 992,
  xl: 1200,
  xxl: 1400
}

export const containerWidths = {
  sm: "540px",
  md: "720px",
  lg: "960px",
  xl: "1140px",
  xxl: "1320px"
}

export const responsive = Object.keys(breakpoints).reduce(
  (accumulator, label) => {
    accumulator[label] = (...args) => css`
      @media (min-width: ${breakpoints[label]}px) {
        ${css(...args)};
      }
    `

    return accumulator
  },
  {}
)

export const font = ({ size, weight, color }) => `
  color: ${color
    ? color
    : colors.black};
  font-size: ${size
    ? size
    : fontSize.nm}px;
  font-family: "NunitoSans-${weight
    ? weight
    : fontWeight.medium}","sans-serif";
`
