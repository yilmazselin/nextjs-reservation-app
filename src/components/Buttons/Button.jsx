import styled, { css } from "styled-components"
import { font } from "../../utils/helper"
import { colors, fontSize, fontWeight } from "../../utils/styles"

const COLOR = {
  primary: css`
    color: #fff;
    background-color: ${colors.red};
  `,
  "outline-primary": css`
    color: ${colors.red};
    background-color: ${colors.white};
    border: 1px solid ${colors.red};
  `,
  secondary: css``
}

const DISABLED = css`
  cursor: default;
  background-color: ${colors.disabled};
  color: ${colors.white};
`

const Button = styled.button`
  width: 100%;
  height: 56px;
  border: none;
  border-radius: 12px;
  outline: none;
  transition: all 0.2s;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  ${font({
    size: fontSize.nm,
    weight: fontWeight.bold,
    color: colors.white
  })};
  ${(props) => props.color && COLOR[props.color]}
  ${(props) => props.disabled && DISABLED};
`

export default Button
