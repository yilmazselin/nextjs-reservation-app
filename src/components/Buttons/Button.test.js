import "@testing-library/jest-dom"
import { fireEvent, render, screen } from "@testing-library/react"
import renderer from "react-test-renderer"
import Button from "./Button"
import "jest-styled-components"
import { colors } from "../../utils/styles"

describe("<Button />", () => {
  it("Should render correctly", () => {
    const component = renderer.create(<Button />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test("Should have background color='primary' when color='primary' prop exists", () => {
    const tree = renderer.create(<Button color="primary" />).toJSON()

    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule("background-color", colors.red)
  })

  test("Should render a disabled button", () => {
    render(<Button children="Disabled" disabled />)
    const button = screen.getByText("Disabled")

    expect(button).toMatchSnapshot()
    expect(button).toHaveStyleRule("background-color", colors.disabled)
    expect(button).toBeDisabled()
  })

  test("button click event test", () => {
    let count = 0
    const setCount = (newCount) => {
      count = newCount
    }

    render(<Button children="Click" onClick={() => setCount(1)} />)
    const button = screen.getByText("Click")

    fireEvent.click(button)
    expect(count = 1)
  })
})
