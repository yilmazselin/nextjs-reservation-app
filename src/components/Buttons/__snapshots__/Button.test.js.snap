// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should have background color='primary' when color='primary' prop exists 1`] = `
.c0 {
  width: 100%;
  height: 56px;
  border: none;
  border-radius: 12px;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #fff;
  font-size: 14px;
  font-family: "NunitoSans-700","sans-serif";
  color: #fff;
  background-color: #E82223;
}

<button
  className="c0"
  color="primary"
/>
`;

exports[`<Button /> Should render a disabled button 1`] = `
.c0 {
  width: 100%;
  height: 56px;
  border: none;
  border-radius: 12px;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #fff;
  font-size: 14px;
  font-family: "NunitoSans-700","sans-serif";
  cursor: default;
  background-color: #BBC3CF;
  color: #fff;
}

<button
  class="c0"
  disabled=""
>
  Disabled
</button>
`;

exports[`<Button /> Should render correctly 1`] = `
.c0 {
  width: 100%;
  height: 56px;
  border: none;
  border-radius: 12px;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #fff;
  font-size: 14px;
  font-family: "NunitoSans-700","sans-serif";
}

<button
  className="c0"
/>
`;
